From ec158d758b880fc966f3f3e6b819ffcfbff7227d Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 14 May 2014 16:38:49 +0200
Subject: [PATCH] mdraid/module-setup.sh: fixed print-cmdline for empty UUID

fixes

/usr/lib/dracut/modules.d/90mdraid/module-setup.sh: line 60:
_activated["${UUID}"]: bad array subscript

Thanks to Jan ONDREJ for spotting this!

(cherry picked from commit 19bb14c3467233f8c98f76b3fa57047a3a31e1a3)
---
 modules.d/90mdraid/module-setup.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules.d/90mdraid/module-setup.sh b/modules.d/90mdraid/module-setup.sh
index 1e03847f..815a3ebc 100755
--- a/modules.d/90mdraid/module-setup.sh
+++ b/modules.d/90mdraid/module-setup.sh
@@ -50,6 +50,8 @@ cmdline() {
             done
         )
 
+        [[ -z "$UUID" ]] && continue
+
         if ! [[ ${_activated[${UUID}]} ]]; then
             printf "%s" " rd.md.uuid=${UUID}"
             _activated["${UUID}"]=1
